### PR TITLE
Fix crash in native crypto on concurrent cipher object use with GCM

### PR DESCRIPTION
**Pull request by ZainabF92**
Porting PR [openj9-openjdk-jdk16#66](https://github.com/ibmruntimes/openj9-openjdk-jdk16/pull/66)

Signed-off-by: Zainab Fatmi <zainab@ibm.com>

### CODE REVIEW

**Comment by ashbm5** [comment]
FYI @pshipton @keithc-ca

File: closed/adds/jdk/src/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java [hunk 1]
```diff
@@ -311,14 +323,16 @@ int decrypt(byte[] cipher, int cipherOffset, int cipherLen,
...
+        int ret;
 
-        if(plain == cipher) {
-            ret = nativeCrypto.CBCFinalEncrypt(nativeContext, plain.clone(),
-                                               plainOffset, plainLen, cipher, cipherOffset);
-        } else {
-            ret = nativeCrypto.CBCFinalEncrypt(nativeContext, plain, plainOffset,
-                                                plainLen, cipher, cipherOffset);
+        synchronized (this) {
+            if(plain == cipher) {
```
Comment (pshipton) [inline closed/adds/jdk/src/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java:329]: Pls add the missing space in `if(`

**Comment by pshipton** [comment]
jenkins test sanity.openjdk alinux64 jdk8

**Comment by pshipton** [comment]
PR build is https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJDK8/17/

File: closed/adds/jdk/src/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java (hunk 1 above)
Comment (ZainabF92) [reply closed/adds/jdk/src/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java]: fixed

**Review by pshipton — APPROVED** [review]

**Pull request merged**